azure/src/test/java/org/apache/iceberg/azure/TestAzureTokenCredentialProviders.java

```diff
@@ -0,0 +1,143 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.azure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import java.time.OffsetDateTime;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+
+public class TestAzureTokenCredentialProviders {
+
+  @Test
+  public void testDefaultFactory() {
+    AzureTokenCredentialProvider provider = AzureTokenCredentialProviders.defaultFactory();
+    assertThat(provider).isNotNull();
+    assertThat(provider)
+        .isInstanceOf(AzureTokenCredentialProviders.DefaultTokenCredentialProvider.class);
+  }
+
+  @Test
+  public void testFromWithNoProvider() {
+    Map<String, String> properties = ImmutableMap.of();
+    AzureTokenCredentialProvider provider = AzureTokenCredentialProviders.from(properties);
+
+    assertThat(provider).isNotNull();
+    assertThat(provider)
+        .isInstanceOf(AzureTokenCredentialProviders.DefaultTokenCredentialProvider.class);
+  }
+
+  @Test
+  public void testFromWithNullProvider() {
+    Map<String, String> properties =
+        ImmutableMap.of(AzureProperties.ADLS_TOKEN_CREDENTIAL_PROVIDER, "");
+    AzureTokenCredentialProvider provider = AzureTokenCredentialProviders.from(properties);
+    assertThat(provider).isNotNull();
+    assertThat(provider)
+        .isInstanceOf(AzureTokenCredentialProviders.DefaultTokenCredentialProvider.class);
+  }
+
+  @Test
+  public void testFromWithDefaultProvider() {
+    Map<String, String> properties =
+        ImmutableMap.of(
+            AzureProperties.ADLS_TOKEN_CREDENTIAL_PROVIDER,
+            "org.apache.iceberg.azure.AzureTokenCredentialProviders$DefaultTokenCredentialProvider");
+    AzureTokenCredentialProvider provider = AzureTokenCredentialProviders.from(properties);
+    assertThat(provider).isNotNull();
+    assertThat(provider)
+        .isInstanceOf(AzureTokenCredentialProviders.DefaultTokenCredentialProvider.class);
+  }
+
+  @Test
+  public void testFromWithDummyProvider() {
+    Map<String, String> properties =
+        ImmutableMap.of(
+            AzureProperties.ADLS_TOKEN_CREDENTIAL_PROVIDER,
+            "org.apache.iceberg.azure.TestAzureTokenCredentialProviders$DummyTokenCredentialProvider");
+    AzureTokenCredentialProvider provider = AzureTokenCredentialProviders.from(properties);
+
+    assertThat(provider).isNotNull();
+    assertThat(provider).isInstanceOf(DummyTokenCredentialProvider.class);
+    assertThat(provider.credential()).isInstanceOf(DummyTokenCredential.class);
```

Comment:
I think we should also update `TestAzureProperties` to make sure the credential actually comes back correctly and has the right value